ledger/internal/cow_creatables.go

```diff
@@ -58,38 +59,106 @@ func (cs *roundCowState) CountAssetParams(addr basics.Address) (int, error) {
 }
 
 func (cs *roundCowState) GetAppParams(addr basics.Address, aidx basics.AppIndex) (ret basics.AppParams, ok bool, err error) {
-	return cs.lookupAppParams(addr, aidx)
+	d, ok, err := cs.lookupAppParams(addr, aidx)
+	if d.Params != nil {
+		ret = *d.Params
+	}
+	if d.Deleted { // XXX return ok=false for deleted?
```

Comment:
check the old implementation:
```
func (cb *roundCowState) lookupAppParams(addr basics.Address, aidx basics.AppIndex) (basics.AppParams, bool, error) {
	params, ok := cb.mods.NewAccts.GetAppParams(addr, aidx)
	if ok {
		if params == nil {
			return basics.AppParams{}, false, nil
		}
		return *params, ok, nil
	}
```
=> ok = false for deleted

Reply:
oh, that's weird, so lookupAppParams is now going to return ok=true for deleted, but the caller here has to change it to ok=false for the implementation of apply.Balances.

Reply:
so I have to audit all the other uses of roundCowState.lookupX? hrmm.

Reply:
maybe lookupAppParams can do the analysis as before?